### PR TITLE
Updated service API

### PR DESCRIPTION
**Pull request by harrisonobeid**

### CODE REVIEW

File: client/src/components/Sidebar.jsx [hunk 1]
```diff
@@ -113,6 +67,6 @@ class Sidebar extends React.Component {
 export default Sidebar;
 
 ReactDOM.render(
-  <Sidebar currentSong="Song_00001" />,
+  <Sidebar currentSong="50" />,
```
Comment (angeliquemari) [inline client/src/components/Sidebar.jsx]: is this supposed to be hardcoded? not this.state.currentSongId?

File: .gitignore [hunk 2]
```diff
@@ -1,2 +1,4 @@
 node_modules
-database/data
\ No newline at end of file
+database/data
+newrelic_agent.log
```
Comment (angeliquemari) [inline .gitignore]: weird, it looks like the file is still getting uploaded (it's in this pr)

File: database/methods.js [hunk 3]
```diff
@@ -4,20 +4,38 @@ const driver = neo4j.driver("bolt://localhost:7687", neo4j.auth.basic("neo4j", "
 
 const getSong = (song_id, res) => {
   var session = driver.session();
-  const cypher = 'MATCH (song:Song {id: {id} }) RETURN song';
   const params = {id: song_id};
-  console.log(params);
-  return session
-    .run(cypher, params)
+  
+  return Promise.all([
```
Comment (angeliquemari) [inline database/methods.js]: just curious - is this a native javascript feature, you didn't need to import bluebird?

File: database/methods.js [hunk 4]
```diff
@@ -4,20 +4,38 @@ const driver = neo4j.driver("bolt://localhost:7687", neo4j.auth.basic("neo4j", "
...
+      res.end();
     })
     .catch(error => {
-      session.close();
+      res.status(500);
+      res.end();
       throw error;
     });
-}
+  }
```
Comment (angeliquemari) [inline database/methods.js]: why did you remove the session.close() statements? (I notice you have them in the read function)

File: database/methods.js [hunk 5]
```diff
@@ -4,20 +4,38 @@ const driver = neo4j.driver("bolt://localhost:7687", neo4j.auth.basic("neo4j", "
...
+    session.run('MATCH (song:Song {id: {id} }) RETURN song.name', params),
+    session.run('MATCH (song:Song {id: {id} })-[:APPEARS_ON]->(album:Album) RETURN album', params)
+  ])
+  .then(result => {
+    res.send(result);
+    session.close();
+  })
+  .catch(error => {
+    session.close();
+    throw error;
```
Comment (angeliquemari) [inline database/methods.js]: You might need to res.end() if there's an error (maybe also set error status)

File: database/methods.js (hunk 3 above)
Comment (harrisonobeid) [reply database/methods.js]: From what I've seen, yes. There's no necessary requirements.

File: database/methods.js (hunk 4 above)
Comment (harrisonobeid) [reply database/methods.js]: Good catch, I should've left that in.

File: database/methods.js (hunk 5 above)
Comment (harrisonobeid) [reply database/methods.js]: Thank you!

File: client/src/index.jsx [hunk 6]
```diff
@@ -4,6 +4,6 @@ import App from './components/App.jsx';
 import 'babel-polyfill';
 
 ReactDOM.render(
-  <App currentSong="Song_00001" />,
+  <App currentSong="388484" />,
```
[the remark this answers is not in the document]
Comment (harrisonobeid) [reply client/src/index.jsx]: In a few days, yes. I just want to ensure that it is functioning correctly first.
